File: src/com/seancheey/MainController.java
package com.seancheey;

import javafx.application.Application;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.TextField;
import javafx.stage.Stage;

import javax.swing.*;

public class MainController extends Application {
    @FXML
    private TextField username, password;

    @Override
    public void start(Stage primaryStage) throws Exception {
        Parent root = FXMLLoader.load(getClass().getResource("sample.fxml"));
        primaryStage.setTitle("Amao Game");
        primaryStage.setScene(new Scene(root, 800, 600));
        primaryStage.show();
    }


    public static void main(String[] args) {
        launch(args);
    }

    public void login() {
        System.out.println(username.getText() + "\n" + password.getText());
    }

    public void register() {

    }

    public void setting() {

    }

    public void credit() {

    }
}

File: src/com/seancheey/scene/controller/MainController.kt
package com.seancheey.scene.controller

/**
 * Created by Seancheey on 27/05/2017.
 * GitHub: https://github.com/Seancheey
 */

import com.seancheey.game.Config
import com.seancheey.game.PlayerSavesReader
import com.seancheey.scene.Scenes
import com.seancheey.scene.Stages
import javafx.fxml.FXML
import javafx.scene.control.Label
import javafx.scene.control.TextField


class MainController {
    @FXML
    private var username: TextField? = null
    @FXML
    private var password: TextField? = null
    @FXML
    private var indicatorLabel: Label? = null

    fun login() {
        if (username!!.text == "") {
            username!!.text = "guest"
        }

        val reader = PlayerSavesReader(username!!.text, password!!.text)
        if (reader.hasSaves) {
            if (reader.passwordCorrect()) {
                Config.player = reader.readPlayer()!!
                Stages.switchScene(Scenes.menu, 800.0, 600.0)
            } else {
                indicatorLabel!!.text = "password not correct"
            }
        } else {
            indicatorLabel!!.text = "username not found, try register"
        }
    }

    fun register() {
        if (username!!.text == "") {
            indicatorLabel!!.text = "username can't be blank"
            return
        }

        // if (password!!.text == "") {
        //    indicatorLabel!!.text = "password can't be blank"
        //    return
        // }
        val reader = PlayerSavesReader(username!!.text, password!!.text)
        if (reader.hasSaves) {
            indicatorLabel!!.text = "username already exists"
            return
        } else {
            Config.player = reader.newPlayer()
            Config.player.saveData()
            indicatorLabel!!.text = "new player created"
            login()
        }
    }

    fun setting() {

    }

    fun credit() {

    }
}
